File: java-algo/src/main/java/com/leetcode/Problem682.java
package com.leetcode;


import java.util.ArrayList;
import java.util.List;

/**
 * 682. Baseball Game
 * https://leetcode.com/problems/baseball-game/
 */
public class Problem682 {
    public int calPoints(String[] ops) {
        List<Integer> q = new ArrayList<>();
        boolean isNeg = false;
        for (String op : ops) {
            switch (op) {
                case "+": {
                    q.add(q.get(q.size() - 1) + q.get(q.size() - 2));
                    break;
                }
                case "D": {
                    q.add(q.get(q.size() - 1) * 2);
                    break;
                }
                case "C":
                    q.remove(q.size() - 1);
                    break;
                case "-":
                    isNeg = true;
                default:
                    int num = Integer.parseInt(op);
                    q.add(isNeg ? -num : num);
                    isNeg = false;
            }
        }
        int ans = 0;
        for (Integer n : q) {
            ans += n;
        }
        return ans;
    }
}

File: python-algo/leetcode/problem_682.py
"""682. Baseball Game
https://leetcode.com/problems/baseball-game/
"""
from typing import List


class Solution:
    def calPoints(self, ops: List[str]) -> int:
        q = []
        neg = False
        for op in ops:
            if op == '+':
                q.append(q[-1] + q[-2])
            elif op == 'D':
                q.append(q[-1] * 2)
            elif op == 'C':
                q.pop()
            elif op == '-':
                neg = True
            else:
                num = int(op)
                q.append(num if not neg else -num)
                neg = False
        return sum(q)
